cmd/tsdb/main.go

```diff
@@ -531,3 +545,32 @@ func analyzeBlock(b *tsdb.Block, limit int) {
 	fmt.Printf("\nHighest cardinality metric names:\n")
 	printInfo(postingInfos)
 }
+
+func dumpSamples(db *tsdb.DB, mint, maxt int64) {
+	q, err := db.Querier(mint, maxt)
+	if err != nil {
+		exitWithError(err)
+	}
+
+	ss, err := q.Select()
```

Comment:
this doesn't work when I tested it. I think you need 

`	ss, err := q.Select(labels.NewMustRegexpMatcher("", ".*"))`

Reply:
Indeed, I thought I had tested it, but I must've accidentally tried it with an old binary that still had a label matcher. Added back that label matcher like you suggested.